x/dex/keeper/msgserver/msg_server_register_pairs.go

```diff
@@ -15,10 +15,14 @@ func (k msgServer) RegisterPairs(goCtx context.Context, msg *types.MsgRegisterPa
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	maxPairsPerContract := k.GetMaxPairsPerContract(ctx)
 	events := []sdk.Event{}
 	// Validation such that only the user who stored the code can register pairs
 	for _, batchPair := range msg.Batchcontractpair {
 		contractAddr := batchPair.ContractAddr
+		if uint64(len(k.GetAllRegisteredPairs(ctx, contractAddr))) >= maxPairsPerContract {
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "contract %s already has %d pairs registered", contractAddr, maxPairsPerContract)
```

Comment:
fixed, and added migration for params. No need to migrate for the new order counter state since it will be overwritten once order book is changed, and empty keys are handled as 0